Sources/AccessibilitySnapshot/Core/Swift/Classes/UIView+ImageRendering.swift

```diff
@@ -0,0 +1,171 @@
+//
+//  Copyright 2023 Block Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import CoreImage
+import UIKit
+
+extension UIView {
+
+    func renderToImage(
+        monochrome: Bool,
+        viewRenderingMode: AccessibilitySnapshotView.ViewRenderingMode
+    ) throws -> UIImage {
+        let renderer = UIGraphicsImageRenderer(bounds: bounds)
+
+        var error: Error?
+
+        let snapshot = renderer.image { context in
+            switch viewRenderingMode {
+            case .drawHierarchyInRect:
+                if bounds.width > UIView.tileSideLength || bounds.height > UIView.tileSideLength {
+                    drawTiledHierarchySnapshots(in: context, error: &error)
+                } else {
+                    drawHierarchy(in: bounds, afterScreenUpdates: true)
+                }
+
+            case .renderLayerInContext:
+                layer.render(in: context.cgContext)
+            }
+        }
+
+        if let error = error {
+            throw error
+        }
+
+        if monochrome {
+            return try monochromeSnapshot(for: snapshot) ?? snapshot
+
+        } else {
+            return snapshot
+        }
+    }
+
+    private func monochromeSnapshot(for snapshot: UIImage) throws -> UIImage? {
+        if ProcessInfo().operatingSystemVersion.majorVersion == 13 {
+            // On iOS 13, the image filter silently fails for large images, "successfully" producing a blank output
```

Comment:
gotta love when xcode fails successfully

Reply:
I don't even understand why it has to here. It could just return a nil outputImage, but nope 🤷‍♂️